components/table-of-contents/src/Section/test.js

```diff
@@ -1,35 +1,71 @@
 import React from 'react';
-import { shallow } from 'enzyme';
+import { mount, shallow } from 'enzyme';
 
-import { Entry } from '.';
+import Section, { Entry } from '.';
 
-describe('TableOfContents Section Entry', () => {
-  let wrapper;
-  const addContent = jest.fn();
-  const removeContent = jest.fn();
-
-  it('renders without crashing', () => {
-    wrapper = shallow(
-      <Entry id="id" title="Title" toc={{ addContent, removeContent }}>
-        Section text here
-      </Entry>,
+let wrapper;
+
+describe('TableOfContents Section', () => {
+  it('renders standalone', () => {
+    wrapper = mount(
+      <Section id="id" title="Title">Section text here</Section>,
     );
+    expect(wrapper.exists()).toBe(true);
   });
+});
 
-  it('has called addContent in context', () => {
-    expect(addContent.mock.calls.length).toBe(1);
-  });
+describe('TableOfContents Section Entry', () => {
+  describe('works with Context API', () => {
+    const addContent = jest.fn();
+    const removeContent = jest.fn();
 
-  it('has not yet called removeContent in context', () => {
-    expect(removeContent.mock.calls.length).toBe(0);
-  });
+    it('renders without crashing', () => {
+      wrapper = shallow(
+        <Entry id="id" title="Title" toc={{ addContent, removeContent }}>
+          Section text here
+        </Entry>,
+      );
+      expect(wrapper.exists()).toBe(true);
+    });
 
-  it('calls removeContent on remove/unmount', () => {
-    wrapper.unmount();
-    expect(removeContent.mock.calls.length).toBe(1);
+    it('has called addContent in context', () => {
+      expect(addContent.mock.calls.length).toBe(1);
```

Comment:
it's better practice in testing for every `it` to be self-contained, in this case the `wrapper` should be generated inside each `it` block separately, with individual definitions of the `addContent` and `removeContent` mocks also inside each block

that way if the testing harness decides to run assertions in parallel, or in a different order, the tests will still pass - additionally tests can be moved around or copied into different `describe` blocks without fear of them breaking

these tests right now rely on code from higher up the file in the `describe` block, which makes them harder to reason about individually

Reply:
Yep, all agreed. As just discussed, these test files are using "legacy format"! I'll update them and fix these things. :)